src/sonic-bgpcfgd/bgpcfgd/managers_advertise_rt.py

```diff
@@ -18,82 +22,140 @@ def __init__(self, common_objs, db, table):
             db,
             table,
         )
-        
-        self.directory.subscribe([("CONFIG_DB", swsscommon.CFG_DEVICE_METADATA_TABLE_NAME, "localhost/bgp_asn"),], self.on_bgp_asn_change)
-        self.advertised_routes = dict()
-
 
-    OP_DELETE = 'DELETE'
-    OP_ADD = 'ADD'
+        self.directory.subscribe(
+            [
+                ("CONFIG_DB", swsscommon.CFG_DEVICE_METADATA_TABLE_NAME, "localhost/bgp_asn"),
+            ],
+            self.on_bgp_asn_change,
+        )
+        self.advertised_routes = dict()
 
+    OP_DELETE = "DELETE"
+    OP_ADD = "ADD"
 
     def set_handler(self, key, data):
+        log_info("AdvertiseRouteMgr:: set handler")
+        if not self._set_handler_validate(key, data):
+            return True
         vrf, ip_prefix = self.split_key(key)
-        self.add_route_advertisement(vrf, ip_prefix)
+        self.add_route_advertisement(vrf, ip_prefix, data)
 
         return True
 
-
     def del_handler(self, key):
+        log_info("AdvertiseRouteMgr:: del handler")
+        if not self._del_handler_validate(key):
+            return
         vrf, ip_prefix = self.split_key(key)
         self.remove_route_advertisement(vrf, ip_prefix)
 
+    def _ip_addr_validate(self, key):
+        if key:
+            _, ip_prefix = self.split_key(key)
+            ip_prefix = ip_prefix.split("/")
+            if len(ip_prefix) != 2:
+                log_err("BGPAdvertiseRouteMgr:: No valid ip prefix for advertised route %s" % key)
+                return False
+            try:
+                ip = ipaddress.ip_address(ip_prefix[0])
+                if ip.version == 4 and int(ip_prefix[1]) not in range(0, 33):
+                    log_err(
+                        "BGPAdvertiseRouteMgr:: ipv4 prefix %s is illegal for advertised route %s" % (ip_prefix[1], key)
+                    )
+                    return False
+                if ip.version == 6 and int(ip_prefix[1]) not in range(0, 129):
+                    log_err(
+                        "BGPAdvertiseRouteMgr:: ipv6 prefix %s is illegal for advertised route %s" % (ip_prefix[1], key)
+                    )
+                    return False
+            except ValueError:
+                log_err("BGPAdvertiseRouteMgr:: No valid ip %s for advertised route %s" % (ip_prefix[0], key))
+                return False
+        else:
+            return False
+        return True
+
+    def _set_handler_validate(self, key, data):
+        if data:
+            if ("profile" in data and data["profile"] in ROUTE_MAPS) or data == {"":""}:
+                return self._ip_addr_validate(key)
```

Comment:
I think, this could be just returning 'true' here. IP prefix should already be validated, and we may just be adding extra delay here.

Reply:
Changed.